crates/bevy_ecs/src/system/query.rs

```diff
@@ -640,6 +679,99 @@ where
         }
     }
 
+    /// Returns the read-only query results for the provided Array of [`Entity`]s.
+    ///
+    /// These values follow the order of your input Array (if anyy).
+    /// In case of a nonexisting entity or mismatched component,
+    /// a [`QueryEntityError`] is returned instead.
+    ///
+    /// If you need to verify the identity of each item returned,
+    /// add [`Entity`] to your [`Query`].
+    ///
+    /// # Example
+    /// ```rust
+    /// # use bevy_ecs::prelude::*;
+    /// #[derive(Component, PartialEq, Debug)]
+    /// struct A(u64);
+    ///
+    /// let mut world = World::new();
+    /// let entity_1 = world.spawn().insert(A(1)).id();
+    /// let entity_2 = world.spawn().insert(A(2)).id();
+    /// let entity_3 = world.spawn().insert(A(3)).id();
+    ///
+    /// let query_state = world.query::<&A>();
+    /// let a_query = Query::from_state(&mut world, &query_state);
```

Comment:
I don't think the QueryState -> Query pattern is something we should encourage in our docs.

Reply:
Query examples should just show systems using the query to avoid the need to show Query construction.

Reply:
Sounds good to me.

Reply:
Doing that isn't too bad right now:

```rust
let mut system = IntoSystem::into_system(some_system);
system.initialize(world);
system.run((), world);
```

And we could / should hide those lines.

And we could add a `run_system(system, world)` function if we really wanted to (that initializes and runs the system now).

Reply:
(and calls IntoSystem)